### ISSUE TITLE
rootio: reading Gate output with root v6.12.04

### ISSUE THREAD

**Issue by kczimm** [go-hep/hep#146]
I'm using [Gate](http://www.opengatecollaboration.org/) for simulations and producing root files. I have installed root v6.12.04. When I try reading the contents of the file using rootio, I get the following dump:

```
kevin@photon:/tmp/rootio$ root-ls -t deposition.root
=== [deposition.root] ===
version: 61204
fatal error: runtime: out of memory

runtime stack:
runtime.throw(0x53b3d3, 0x16)
        /usr/local/go/src/runtime/panic.go:619 +0x81
runtime.sysMap(0xc420100000, 0x1aa130000, 0xc400000000, 0x616cd8)
        /usr/local/go/src/runtime/mem_linux.go:216 +0x20a
runtime.(*mheap).sysAlloc(0x5fe640, 0x1aa130000, 0x7f7d9b2dd600)
        /usr/local/go/src/runtime/malloc.go:470 +0xd4
runtime.(*mheap).grow(0x5fe640, 0xd5098, 0x0)
        /usr/local/go/src/runtime/mheap.go:907 +0x60
runtime.(*mheap).allocSpanLocked(0x5fe640, 0xd5098, 0x616ce8, 0x7ffcc8bfd470)
        /usr/local/go/src/runtime/mheap.go:820 +0x301
runtime.(*mheap).alloc_m(0x5fe640, 0xd5098, 0x410100, 0xc41fff87ff)
        /usr/local/go/src/runtime/mheap.go:686 +0x118
runtime.(*mheap).alloc.func1()
        /usr/local/go/src/runtime/mheap.go:753 +0x4d
runtime.(*mheap).alloc(0x5fe640, 0xd5098, 0xc420010100, 0x5f9ac0)
        /usr/local/go/src/runtime/mheap.go:752 +0x8a
runtime.largeAlloc(0x1aa12f100, 0x610001, 0x7f7d9b27c540)
        /usr/local/go/src/runtime/malloc.go:826 +0x94
runtime.mallocgc.func1()
        /usr/local/go/src/runtime/malloc.go:721 +0x46
runtime.systemstack(0x0)
        /usr/local/go/src/runtime/asm_amd64.s:409 +0x79
runtime.mstart()
        /usr/local/go/src/runtime/proc.go:1170

goroutine 1 [running]:
runtime.systemstack_switch()
        /usr/local/go/src/runtime/asm_amd64.s:363 fp=0xc420049328 sp=0xc420049320 pc=0x44f8a0
runtime.mallocgc(0x1aa12f100, 0x50f620, 0x540000c420049401, 0x15)
        /usr/local/go/src/runtime/malloc.go:720 +0x8a2 fp=0xc4200493c8 sp=0xc420049328 pc=0x40f642
runtime.makeslice(0x50f620, 0x1aa12f10, 0x1aa12f10, 0xc420049450, 0x4bd6f7, 0xc42000e960)
        /usr/local/go/src/runtime/slice.go:61 +0x77 fp=0xc4200493f8 sp=0xc4200493c8 pc=0x43bc37
go-hep.org/x/hep/rootio.(*objarray).UnmarshalROOT(0xc420049568, 0xc42000e960, 0x0, 0x0)
        /home/kevin/go/src/go-hep.org/x/hep/rootio/objarray.go:67 +0xb4 fp=0xc420049460 sp=0xc4200493f8 pc=0x4d4834
go-hep.org/x/hep/rootio.(*ttree).UnmarshalROOT(0xc4200f0000, 0xc42000e960, 0xc4200f0000, 0x7f7d9b27c540)
        /home/kevin/go/src/go-hep.org/x/hep/rootio/tree.go:160 +0x401 fp=0xc420049668 sp=0xc420049460 pc=0x4df1c1
go-hep.org/x/hep/rootio.(*Key).Object(0xc420049b60, 0x0, 0x0, 0x0, 0xc42000e8d0)
        /home/kevin/go/src/go-hep.org/x/hep/rootio/key.go:113 +0x221 fp=0xc4200497c8 sp=0xc420049668 pc=0x4ca9d1
go-hep.org/x/hep/rootio.(*Key).Value(0xc420049b60, 0x49206c, 0xc420014269)
        /home/kevin/go/src/go-hep.org/x/hep/rootio/key.go:79 +0x2b fp=0xc420049800 sp=0xc4200497c8 pc=0x4ca74b
main.walk(0x552a00, 0xc420086100, 0xc42008a000, 0x40000139a, 0x5cdb, 0x0, 0xed2263015, 0x0, 0xa0000003c, 0xc39cf, ...)
        /home/kevin/go/src/go-hep.org/x/hep/rootio/cmd/root-ls/main.go:131 +0x43 fp=0xc420049b50 sp=0xc420049800 pc=0x4ea373
main.main()
        /home/kevin/go/src/go-hep.org/x/hep/rootio/cmd/root-ls/main.go:124 +0x132 fp=0xc420049f88 sp=0xc420049b50 pc=0x4e9812
runtime.main()
        /usr/local/go/src/runtime/proc.go:198 +0x212 fp=0xc420049fe0 sp=0xc420049f88 pc=0x429582
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:2361 +0x1 fp=0xc420049fe8 sp=0xc420049fe0 pc=0x452321
```

Is this version of root not supported? Or does anyone know if Gate v8.0 creates a root file unreadable by rootio?

The file I'm trying to read is less than 1 MB and my system has 8 GB memory, however the runtime error out of memory is thrown.

Thanks in advance for the insight.

**Comment by sbinet** [comment]
Sorry about that...
Could you put the file somewhere I could have a look at?

**Comment by sbinet** [comment]
(also, please make sure you are using 'v0.10' as this is the version that got initial support for the ROOT version you're using. From the stack trace, it seems the line numbers in tree.go would match something pre-v0.10.0)

**Comment by kczimm** [comment]
Ah, I'm sorry. I must not have had the most recent version of rootio. I upgraded and it seems to work now. Thanks for your help, and sorry for the inconvenience.

**Comment by sbinet** [comment]
no worries!
I am glad I was able to go back in time and fix your bug in `v0.10` :}
